packages/fee-estimation/README.md

```diff
@@ -0,0 +1,112 @@
+# GasEstimation Documentation
+
+## Why
+
+Gas estimation on OP-chains has both an l2 and l1 component. By default tools such as Viem, Wagmi, Ethers, and Web3.js do not support the l1 component. They will support this soon but in meantime this library can help estimate fees for transactions.
+As these tools add support for gas estimation natively this README will be updated with framework specific instructions.
+
+## High level
+
+- The l2 contract that can estimate l1Fees is called [GasPriceOracle](../contracts-bedrock/contracts/l2/GasPriceOracle.sol) contract. This library provides utils for interacting with it at a high level.
+- The GasPriceOracle is [deployed to Optimism](https://optimistic.etherscan.io/address/0x420000000000000000000000000000000000000F) and other OP chains at a predeployed address of `0x420000000000000000000000000000000000000F`
+
+For more detailed information about gas fees on Optimism's Layer 2, you can visit their [official documentation](https://community.optimism.io/docs/developers/build/transaction-fees/#the-l2-execution-fee).
+
+## Installation:
+
+```bash
+pnpm i @eth-optimism/fee-estimation
+```
+
+```bash
+npm i @eth-optimism/fee-estimation
+```
+
+```
+yarn add @eth-optimism/fee-estimation
+```
+
+## Importing the Gas Fee Estimator
+
+To use the Gas Fee Estimator in your file, import it:
+
+```typescript
+import {
+  baseFee,
+  decimals,
+  estimateFees,
+  gasPrice,
+  getL1Fee,
+  getL1GasUsed,
+  l1BaseFee,
+  overhead,
+  scalar,
+  version,
+} from '@eth-optimism/fee-estimator'
+```
+
+## Usage
+
+### Provider Options
+
+Provider options are required for most function calls. They specify the chain and RPC URL for fetching data. They can be defined as follows:
+
+```typescript
+const chainConfig = {
+  chainId: 10,
+  rpcUrl: 'https://mainnet.optimism.io',
+}
+```
+
+### Block Options
+
+Block options specify the block from which to retrieve data. They can be defined as follows:
+
+```typescript
+const blockConfig = {
+  blockNumber: BigInt(106889079),
+}
```

Comment:
same Q as above, not a blocker but just wondering if configuring this will mean more work on the average case

Reply:
This is an optional config nobody would usually use. This is only for when you are querying a past blockNumber.  I use it specifically in tests because using it in all my tests makes it so the tests are very deterministic unlikely to break in future since they always run against the same state.  99% of users would just leave this blank which tells viem to use the default `latest` blocktag

Reply:
calling out the optional-ness would be nice! i thought it wasn't on first read